src/libraries/System.Linq.Queryable/src/System/Linq/Queryable.cs

```diff
@@ -168,7 +168,7 @@ public static IQueryable<TResult> Cast<TResult>(this IQueryable source)
 
         private static Expression GetSourceExpression<TSource>(IEnumerable<TSource> source)
         {
-            IQueryable<TSource> q = source as IQueryable<TSource>;
+            IQueryable<TSource>? q = source as IQueryable<TSource>;
             return q != null ? q.Expression : Expression.Constant(source, typeof(IEnumerable<TSource>));
         }
 
```

Comment:
Seems lots of implementations here depend on Linq.Expression, as disscussed offline gonna wait for that one annotated first